lib/fastly_nsq/messenger.rb

```diff
@@ -4,12 +4,12 @@ module FastlyNsq::Messenger
 
   module_function
 
-  def deliver(message:, on_topic:, originating_service: nil)
+  def deliver(message:, on_topic:, originating_service: nil, meta: {})
     payload = {
       data: message,
       meta: {
         originating_service: originating_service || self.originating_service,
-      },
+      }.merge(meta),
```

Comment:
could do
```
meta[:originating_service] = originating_service || self.originating_service
payload =  {
  data: message,
  meta: meta
}
```

Reply:
or you could raise `NotAllowedToUpdateThatError` :P